src/__test__/useConnection.test.ts

```diff
@@ -1,77 +1,79 @@
 import { renderHook, act } from "@testing-library/react-hooks";
-import Onboard from "bnc-onboard";
-import * as ethers from "ethers";
-import {
-  API as OnboardApi,
-  Initialization,
-} from "bnc-onboard/dist/src/interfaces";
-import EventEmitter from "events";
 
-import { useConnection, ConnectionProvider } from "../hooks";
+import { ethers } from "ethers";
 
-jest.mock("bnc-onboard");
-const mockedOnboard = Onboard as jest.Mock<Partial<OnboardApi>>;
-const mockedWalletSelect = jest.fn();
-const mockedWalletCheck = jest.fn();
+import EventEmitter from "events";
 
-const mockProviderFn: ethers.providers.JsonRpcFetchFunc = jest.fn();
+import { useConnection, ConnectionProvider } from "../hooks/useConnection";
 
-jest
-  .spyOn(ethers.providers.Web3Provider.prototype, "getNetwork")
-  .mockImplementation(async () => {
-    return {
-      chainId: 1,
-      name: "mainnet",
-    };
-  });
-let fakeProvider: ethers.providers.Web3Provider;
-let fakeNetwork: ethers.providers.Network;
-let fakeAddress: string;
+const provider = new ethers.providers.JsonRpcProvider("http://localhost:8545");
+const jsonRpcFunc = (method: string, params?: Array<any>) =>
+  provider.send(method, params || []);
+const web3Provider = new ethers.providers.Web3Provider(jsonRpcFunc);
+const wallet = ethers.Wallet.createRandom();
+const randomAddress = ethers.Wallet.createRandom().address;
+
+class Connector extends EventEmitter {
+  provider: ethers.providers.Web3Provider;
+  signer: ethers.Signer;
+  constructor(provider: ethers.providers.Web3Provider, signer: ethers.Signer) {
+    super();
+    this.provider = provider;
+    this.signer = signer;
+  }
+  emitUpdate(update: any) {
+    this.emit("update", update);
+  }
+  emitChainChanged(chainId: number) {
+    this.emit("chain_changed", chainId);
+  }
+  emitAccountChanged(account: string) {
+    this.emit("account_changed", account);
+  }
+}
```

Comment:
i dont think this class is necessary. you arent using the emit functions since its easy enough to just call emit directly in the test.  After further thought, you can just simplify a lot by calling the useConnection hook functions directly in the tests and making sure the state gets set, as opposed to writing a wrapper class that indirectly calls them through events.

Reply:
Yeah I think you're right. I was trying to reproduce a "real" situation but its def unnecessary for this test